lang/java/tools/src/main/java/org/apache/avro/tool/ConcatTool.java

```diff
@@ -74,17 +99,8 @@ public int run(InputStream in, PrintStream out, PrintStream err,
         // this is the first file - set up the writer, and store the
         // Schema & metadata we'll use.
         schema = reader.getSchema();
-        for (String key : reader.getMetaKeys()) {
-          if (!DataFileWriter.isReservedMeta(key)) {
-            byte[] metadatum = reader.getMeta(key);
-            metadata.put(key, metadatum);
-            writer.setMeta(key, metadatum);
-          }
-        }
-        inputCodec = reader.getMetaString(DataFileConstants.CODEC);
-        if(inputCodec == null) {
-          inputCodec = DataFileConstants.NULL_CODEC;
-        }
+        extractMetadata(reader, metadata, writer);
```

Comment:
Is there anything in particular that should be tested with this? Are the meta entries just a dictionary of random stuff?